go.mod

```diff
@@ -1,7 +1,6 @@
 module github.com/cilium/cilium
 
-// renovate: datasource=golang-version depName=go
```

Comment:
I think we can keep this renovate line and add the following:
```
$ git diff
diff --git a/.github/renovate.json5 b/.github/renovate.json5
index a1544a26f8..8170b98d68 100644
--- a/.github/renovate.json5
+++ b/.github/renovate.json5
@@ -119,6 +119,26 @@
         "main"
       ]
     },
+    {
+      "groupName": "only major minor golang",
+      "groupSlug": "only-major-minor-golang",
+      "matchFiles": [
+        "go.mod",
+      ],
+      "matchDepNames": [
+        "go"
+      ],
+      "matchDatasources": [
+        "golang-version"
+      ],
+      "matchUpdateTypes": [
+        "major",
+        "minor",
+      ],
+      matchBaseBranches: [
+        "main"
+      ]
+    },
     {
       // Do not allow any updates into stable branches.
       "enabled": false,
```

Reply:
SGTM. Does this generate `1.21.0` as output?

Reply:
Oh incidentally, I just took this patch verbatim and applied it... How would one test this?

Reply:
> SGTM. Does this generate 1.21.0 as output?

You can test it locally with https://docs.renovatebot.com/modules/platform/local/

Reply:
LGTM, as long as the lack of GitHub token doesn't break local testing:

```
$ docker run -ti -v $(pwd):/usr/src/app docker.io/renovate/renovate:full renovate --platform=local
 INFO: Repository started (repository=local)
       "renovateVersion": "37.31.1"
fatal: detected dubious ownership in repository at '/usr/src/app'
To add an exception for this directory, call:

        git config --global --add safe.directory /usr/src/app
fatal: detected dubious ownership in repository at '/usr/src/app'
To add an exception for this directory, call:

        git config --global --add safe.directory /usr/src/app
fatal: detected dubious ownership in repository at '/usr/src/app'
To add an exception for this directory, call:

        git config --global --add safe.directory /usr/src/app
fatal: detected dubious ownership in repository at '/usr/src/app'
To add an exception for this directory, call:

        git config --global --add safe.directory /usr/src/app
fatal: detected dubious ownership in repository at '/usr/src/app'
To add an exception for this directory, call:

        git config --global --add safe.directory /usr/src/app
fatal: detected dubious ownership in repository at '/usr/src/app'
To add an exception for this directory, call:

        git config --global --add safe.directory /usr/src/app
 INFO: Dependency extraction complete (repository=local)
       "stats": {
         "managers": {
           "dockerfile": {"fileCount": 9, "depCount": 31},
           "github-actions": {"fileCount": 49, "depCount": 496},
           "gomod": {"fileCount": 1, "depCount": 252},
           "kubernetes": {"fileCount": 1, "depCount": 2},
           "regex": {"fileCount": 42, "depCount": 121}
         },
         "total": {"fileCount": 102, "depCount": 902}
       }
 WARN: GitHub token is required for some dependencies (repository=local)
       "githubDeps": [
         "actions/checkout",
         "docker/setup-buildx-action",
         "docker/setup-qemu-action",
         "sigstore/cosign-installer",
         "docker/login-action",
         "docker/build-push-action",
         "actions/upload-artifact",
         "cilium/actions-app-token",
         "actions/download-artifact",
         "actions/cache",
         "actions/stale",
         "myrotvorets/set-commit-status-action",
         "cilium/cilium-cli",
         "azure/login",
         "aanm/junit2md",
         "aws-actions/configure-aws-credentials",
         "helm/kind-action",
         "cilium/little-vm-helper",
         "google-github-actions/auth",
         "google-github-actions/setup-gcloud",
         "nick-invision/retry",
         "actions/setup-go",
         "cilium/cilium",
         "anchore/scan-action",
         "dorny/paths-filter",
         "actions/github-script",
         "KyleMayes/install-llvm-action",
         "github/codeql-action",
         "golangci/golangci-lint-action",
         "actions/setup-python",
         "hramos/needs-attention",
         "cilium/reusable-workflows",
         "google/oss-fuzz",
         "cilium/hubble",
         "google/gops",
         "kubernetes-sigs/bom",
         "eksctl-io/eksctl",
         "kubernetes/kubernetes",
         "kubernetes-sigs/kind",
         "kubernetes/kops"
       ]
 INFO: Repository finished (repository=local)
       "cloned": undefined,
       "durationMs": 171608
```

Reply:
I have no idea what this output means. I managed to get a local copy built and running and it got rid of all the warnings (`$GITHUB_COM_TOKEN`) but it's not really telling me anything. I can't tell if that means:
- Config is good, success because there's no errors
- Config is bad, command succeeds but the configuration is ineffectual.

```
cilium $ renovate --platform=local
 INFO: Repository started (repository=local)
       "renovateVersion": "0.0.0-semantic-release"
 INFO: Dependency extraction complete (repository=local)
       "stats": {
         "managers": {
           "dockerfile": {"fileCount": 9, "depCount": 31},
           "github-actions": {"fileCount": 49, "depCount": 499},
           "gomod": {"fileCount": 1, "depCount": 254},
           "kubernetes": {"fileCount": 1, "depCount": 2},
           "regex": {"fileCount": 42, "depCount": 122}
         },
         "total": {"fileCount": 102, "depCount": 908}
       }
 INFO: Repository finished (repository=local)
       "cloned": undefined,
       "durationMs": 30337
```

If I use `--dry-run=full` then it fails out with what looks like an upstream renovate bug:

```
cilium $ renovate --platform=local --dry-run=full
 INFO: Repository started (repository=local)
       "renovateVersion": "0.0.0-semantic-release"
 INFO: Dependency extraction complete (repository=local)
       "stats": {
         "managers": {
           "dockerfile": {"fileCount": 9, "depCount": 31},
           "github-actions": {"fileCount": 49, "depCount": 499},
           "gomod": {"fileCount": 1, "depCount": 254},
           "kubernetes": {"fileCount": 1, "depCount": 2},
           "regex": {"fileCount": 42, "depCount": 122}
         },
         "total": {"fileCount": 102, "depCount": 908}
       }
 INFO: DRY-RUN: Would ensure Dependency Dashboard (repository=local)
       "title": "Dependency Dashboard"
 INFO: DRY-RUN: Would save repository cache. (repository=local)
ERROR: Repository has unknown error (repository=local)
       "err": {
         "message": "Cannot convert undefined or null to object",
         "stack": "TypeError: Cannot convert undefined or null to object\n    at Function.keys (<anonymous>)\n    at getBranchList (/home/joe/git/renovate/lib/util/git/index.ts:576:17)\n    at pruneStaleBranches (/home/joe/git/renovate/lib/workers/reposit
ory/finalize/prune.ts:112:39)\n    at finalizeRepo (/home/joe/git/renovate/lib/workers/repository/finalize/index.ts:21:27)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at Object.renovateRepository (/home/joe/git/renovate
/lib/workers/repository/index.ts:94:7)\n    at attributes.repository (/home/joe/git/renovate/lib/workers/global/index.ts:184:11)\n    at start (/home/joe/git/renovate/lib/workers/global/index.ts:169:7)\n    at /home/joe/git/renovate/lib/renovate.ts:18:22"
       }
 INFO: Repository finished (repository=local)
       "cloned": undefined,
       "durationMs": 47498
 INFO: Renovate is exiting with a non-zero code due to the following logged errors
       "loggerErrors": [
         {
           "name": "renovate",
           "level": 50,
           "logContext": "rs4MDLc4tifxh-rQL9lMT",
           "repository": "local",
           "err": {
             "message": "Cannot convert undefined or null to object",
             "stack": "TypeError: Cannot convert undefined or null to object\n    at Function.keys (<anonymous>)\n    at getBranchList (/home/joe/git/renovate/lib/util/git/index.ts:576:17)\n    at pruneStaleBranches (/home/joe/git/renovate/lib/workers/rep
ository/finalize/prune.ts:112:39)\n    at finalizeRepo (/home/joe/git/renovate/lib/workers/repository/finalize/index.ts:21:27)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at Object.renovateRepository (/home/joe/git/reno
vate/lib/workers/repository/index.ts:94:7)\n    at attributes.repository (/home/joe/git/renovate/lib/workers/global/index.ts:184:11)\n    at start (/home/joe/git/renovate/lib/workers/global/index.ts:169:7)\n    at /home/joe/git/renovate/lib/renovate.ts:18
:22"
           },
           "msg": "Repository has unknown error"
         }
       ]
```

Reply:
OK, renovate fails the same way with `--dry-run=full` on tip of main so that gives no signal.

Reply:
I feel like I'm no further to understanding whether the config is correct or not. We still have to just YOLO merge it and hope it's correct?

Reply:
How did you set up renovate on a private fork? My account doesn't show up under the marketplace search box if I try to add it to my personal account: https://github.com/marketplace/renovate .

Reply:
Aha, that link doesn't allow personal accounts. But this link does: https://github.com/apps/renovate . I've kicked off a local test.

Reply:
OK, tested and the new version works as expected - basically the other existing rule _enables_ all go updates, so we need another rule than then disables patches on top.

Reply:
Tested by going back to 1.20.0 => Renovate proposes 1.21.x . Also reverting to 1.20.0 as per this PR, renovate does not propose an update.